### PR TITLE
PluginFactory exposes config.

### PR DESCRIPTION
**Pull request by d3dc**
Rematch initializes PluginFactory with a config

### CODE REVIEW

File: src/pluginFactory.ts [hunk 1]
```diff
@@ -6,7 +6,9 @@ import validate from './utils/validate'
  *
  * makes Plugin objects extend and inherit from a root PluginFactory
  */
-export default () => ({
+export default (config: any) => ({
```
Comment (ShMcK) [inline src/pluginFactory.ts]: In this case the type would be `R.Config`

**Comment by ShMcK** [comment]
Unfortunately, this PR is breaking TypeScript in some horrible ways that will take me a while to wrap my head around :) PluginFactory inherits from Plugin.

**Comment by d3dc** [comment]
... what if it was just `rematchConfig`? The typings would be a little
confusing.

For my use, the store name PR is working.

**Comment by ShMcK** [comment]
TypeScript and I have a love hate relationship. I'm sorry you had to come in the middle of this. We'll sort it out, we also do.

**Pull request merged**